src/assisted_test_infra/test_infra/controllers/node_controllers/oci_api_controller.py

```diff
@@ -460,9 +468,16 @@ def prepare_nodes(self) -> None:
             base_dns=self._entity_config.base_dns_domain,
         )
         stack_id = self._create_stack(
-            random_name("stack-"), namespace, bucket_name, self._config.oci_infrastructure_zip_file, terraform_variables
+            f"stack-{self._entity_config.cluster_id}",
+            namespace,
+            bucket_name,
+            self._config.oci_infrastructure_zip_file,
+            terraform_variables,
         )
         terraform_output = self._apply_job_from_stack(stack_id, random_name("job-"))
```

Comment:
```suggestion
        terraform_output = self._apply_job_from_stack(stack_id, "job-{self._entity_config.cluster_id}")
```
as we applied it only once, it should be enough.

Reply:
For my understanding QE has sometimes more then one job per cluster, that is why left it as it is. 
@talhil-rh @dudyas6 Can you reffer to that?

Reply:
the job is attached to the stack